packages/webapp/src/webview/ui/components/Header/NavigationButtons/NavigationButtons.tsx

```diff
@@ -94,13 +94,19 @@ export function ShareButton() {
     const treeId = useSelector<AppState, GuidedAnswerTreeId>((state) => state.activeGuidedAnswer!?.TREE_ID);
     const nodes = useSelector<AppState, GuidedAnswerNode[]>((state) => state.activeGuidedAnswerNode);
     const [link, setLink] = useState('');
+    const [webLink, setWebLink] = useState('');
 
     useEffect(() => {
         setLink(
             `vscode://saposs.sap-guided-answers-extension#/tree/${treeId}/actions/${nodes
                 .map((n) => n.NODE_ID)
                 .join(':')}`
         );
+        setWebLink(
+            `https://ga.support.sap.com/dtp/viewer/index.html#/tree/${treeId}/actions/${nodes
```

Comment:
Hi @oliversberzs (and @unseen1980),
While the api host `ga.support.sap.com` is correct in most cases, it can be changed using setting `sap.ux.guidedAnswer.apiHost`. I'm also not in favor that the UI does the link generation, also the `vscode:// ...` link is hard-coded, while, depending on the used version it could also be `vscode-insiders:// ...`. Here is my proposal:

We enhance the global state by share information, e.g.:
```typescript
export interface AppState {
    ...
    activeNodeSharing?: {
        extensionLink: string;
        webLink: string;
    };
}
```

We could use `activeNodeSharing` as indication whether the sharing callout is open or not, which would also eliminate the need for local state `isCalloutVisible`, `link`, `webLink` in `NavigationButtons.tsx`. When the share button is clicked, we fire an action which causes the `activeNodeSharing` object to be set in `guidedAnswersPanel.ts`. For extension link there is already a prepared function: 
https://github.com/SAP/guided-answers-extension/blob/d23b204b80f297f8aee21561e384cd65e6e6acc1/packages/ide-extension/src/links/link-info.ts#L41-L45

We would need to add another function for generating the web link. 
This way we would have all the information to generate the link available (api host and vscode/vscode-insiders) and the UI would be kept clean of logic to calculate the link.

What do you think?

If the proposal is not clear, I can create a small branch to showcase what I meant.

Reply:
yeah, the proposal makes sense. 

the only thing I would disagree with is I think the `isCalloutVisible` state should be local so that we don't have to track it through refreshes and can have one solution and one variable in case we need another button with a callout, so each doesn't have a global state variable for it. 

in that case we could generate both links once each time the node is switched, not every  single time the callout is opened.

what do you think?

Reply:
If we add sharing information to the global state anyway, I thought we could just use this information as indicator whether the callout is visible or not. As it becomes difficult to explain, I've branched of `feat/426/web-link` and created a PR to showcase my proposal: https://github.com/SAP/guided-answers-extension/pull/467/files
It has no tests yet, but contains the two new actions, state additions and removes local states that would be not required anymore. Please have a look and let me know what you think.

Reply:
I've looked over the proposal. I understand what you mean, but I still think that it does not make sense to generate the links every time the user clicks on the popup button, when we can generate them in the `SELECT_NODE` event instead. and I think separating the boolean `isCalloutOpen` (in whatever way we want, global or local state) from the links would be a lot more clear what the purpose of it is, for somebody new to the codebase, instead of having it null or non-null, which could logically also mean that "a link to this node does not exist" etc.

Reply:
>but I still think that it does not make sense to generate the links every time the user clicks on the popup button, when we can generate them in the SELECT_NODE event instead

Why? In `SELECT_NODE` the links are generated every time, in case the callout is opened it will be generated only if we potentially need it. Also, if we want to move closer to the philosophy of react toolkit, where the state is sliced and reducers should not change other parts of the state it would be difficult and not so clean to update `activeNodeSharing` in `SELECT_NODE`. 

>and I think separating the boolean isCalloutOpen (in whatever way we want, global or local state) from the links would be a lot more clear what the purpose of it is, for somebody new to the codebase, instead of having it null or non-null, which could logically also mean that "a link to this node does not exist" etc.

Yes, I agree and I'm completely fine with having `activeNodeSharing` not optional but required on `AppState`, like:
```
export interface AppState {
   ...
    activeNodeSharing: ShareNodeLinks;
```

and adding `isCalloutOpen` to the interface:

```
export interface ShareNodeLinks {
    isCalloutOpen: boolean
    extensionLink: string;
    webLink: string;
}
```

Regarding global versus local state, this is often discussion and there are reasons and arguments for both. I'm clearly in favor of having as much as possible in the globals state, local states should be the exception in my opinion. For similar reasons, this is also why this app is using react functional components over class components. Reasons for a globals state include:
- being able to have a fully serializable and controlled version of the application at all times (restoring the state of an application is not possible with local states)
- local state is harder to test
- easy to track all actions in a central place (the reducer). This is especially useful for logging and telemetry. E.g. it would be hard to track how often the share dialog is opened when using local state that does not use reducers.

Perhaps you can change the branch as you like it to have and I can review again. Feel also free to merge https://github.com/SAP/guided-answers-extension/pull/467 into this branch and adjust it with the changes we've discussed.

Reply:
sounds good!